vars/waitForHelixRuns.groovy

```diff
@@ -38,7 +39,27 @@ def call (def helixRunsBlob, String prStatusPrefix) {
                     assert statusResponse.content != null
                     def statusContent = (new JsonSlurperClassic()).parseText(statusResponse.content)
                     def mcResultsUrl = "https://mc.dot.net/#/user/${getEncodedUrl(statusContent.Creator)}/${getEncodedUrl(statusContent.Source)}/${getEncodedUrl(statusContent.Type)}/${getEncodedUrl(statusContent.Build)}"
-                    mcUrlMap[queueId] = mcResultsUrl
+
+                    // Attempt to use the property bag (arch and config), if available, to make the helix keys to the task map more descriptive.
+                    helixRunKeys[correlationId] = queueId
+
+                    if (statusContent.Properties != null) {
+                        if (statusContent.Properties.architecture != null) {
+                            helixRunKeys[correlationId] += " - ${statusContent.Properties.architecture}"
+                        }
+                        if (statusContent.Properties.configuration != null) {
+                            helixRunKeys[correlationId] += " - ${statusContent.Properties.configuration}"
+                        }
+                    }
+
+                    // Find a dupe if one exists.  If it exists, append the correlation id.
+                    def dupe = helixRunKeys.find { it.value == helixRunKeys[correlationId] }
```

Comment:
I believe this always finds itself ?

Reply:
ya not done here yet..I want to augment this to ensure that the dedupe is symmetrical